src/Build/Collections/WeakReference.cs

```diff
@@ -39,7 +34,7 @@ internal class WeakReference<T>
         /// </summary>
         internal WeakReference(T target)
```

Comment:
There's been a [Framework `WeakReference<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.weakreference-1?view=netframework-4.7.1#applies-to) since 4.5. Can we switch to that and just delete this?

Reply:
Removed. Audited all WeakReference usage in the codebase. OpportunisticIntern.cs could really benefit from WeakReference<T> but because it's in Shared it can't use it until the last of the .NET 3.5 code back compat is removed. I did update OpportunisticIntern.cs in the usual ways; please review after the next push.